### PR TITLE
Screen position supplier creation

### LINKED ISSUE #1064: Issue when using double screen
With the ScreenDimensionSupplier we are always taking the dimension of the Screen.getMainScreen(). But maybe the user wants to play GazePlay on another screen, then the xy position of the gaze is wrongly computed.

### LINKED ISSUE #1067: Bad implementation of positionX, positionY of gaze with Tobii
Implementation of positionX, positionY in `poll()` function of `PositionPollerRunnable.java` leads to : #1064
The **_absolute coordinate_** of the gaze have to be return, not the coordinate **_relative to main_** screen.

Then, wen calling `eventFire` in `tobiiGazeDeviceManager.onGazeUpdate` we are firing event with a **_screen position_**, when we should fire event with **_local position_** (like mouseEvent) , leading to #624.

### PR DESCRIPTION
**Pull request by kuhlkrein**
In addition to the ScreenDimensionSupplier, the ScreenPositionSupplier solves the problems of offset related to the use of multiple screen. fix #1067 and fix #1064

### CODE REVIEW

**Comment by kuhlkrein** [comment]
@YannRobert I tested all the game and fixed the ones that had issues related to the changes in the GazeEvent, it is fixing #1067 and #1064  that are both related to the location returned by our _PositionPollerRunnable.poll()_ function. With this PR the way of using event.getX() for MouseEvent and GazeEvent should now be the same. I inspired the implementation of the screenPositionSupplier with your screenDimensionSupplier to use it the same way. Any suggestion ?

**Comment by YannRobert** [comment]
Sounds good
I'll look at the change set in the weekend

File: gazeplay/src/main/java/net/gazeplay/CurrentScreenPositionSupplierFactoryBean.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+public class CurrentScreenPositionSupplierFactoryBean implements FactoryBean<Supplier<Point2D>> {
+
+    @Autowired
+    private GazePlay gazePlay;
+
+    @Override
+    public Supplier<Point2D> getObject() {
+        return new CachingSupplier<>(
+            new ScreenPositionSupplier(
+                new CurrentScreenSupplier(gazePlay)
```
Comment (YannRobert) [inline gazeplay/src/main/java/net/gazeplay/CurrentScreenPositionSupplierFactoryBean.java]: There is already an instance of CurrentScreenSupplier
so we should probably share the same instance

Maybe not a big deal though

File: gazeplay-games/src/main/java/net/gazeplay/games/rushhour/Car.java [hunk 2]
```diff
@@ -153,16 +140,11 @@ public void update(final int size) {
     }
 
     private int checkPos(final Point mouse) {
```
Comment (YannRobert) [inline gazeplay-games/src/main/java/net/gazeplay/games/rushhour/Car.java]: technically the Point is not ```mouse``` only, it maybe gaze. So probably should be renamed.

File: gazeplay-games/src/main/java/net/gazeplay/games/colors/ColorToolBox.java [hunk 3]
```diff
@@ -467,9 +469,8 @@ private Pane buildColorizationPane() {
             getColorsGame().getGameContext().getGazeDeviceManager());
 
         stopColorizeButtonPane.setVisible(false);
-        root.getChildren().add(colorizeButtonIndicator);
 
-        return new StackPane(colorizeButtonPane, stopColorizeButtonPane);
+        return new StackPane(colorizeButtonPane, stopColorizeButtonPane,colorizeButtonIndicator );
```
Comment (YannRobert) [inline gazeplay-games/src/main/java/net/gazeplay/games/colors/ColorToolBox.java]: reformat code

File: gazeplay-games/src/main/java/net/gazeplay/games/colors/ColorToolBox.java [hunk 4]
```diff
@@ -438,7 +439,8 @@ private Pane buildColorizationPane() {
             stopColorize = new Button("S");
         }
 
-        final AbstractGazeIndicator colorizeButtonIndicator = new GazeFollowerIndicator(gameContext, root);
+        final AbstractGazeIndicator colorizeButtonIndicator = new AbstractGazeIndicator(gameContext);
```
Comment (YannRobert) [inline gazeplay-games/src/main/java/net/gazeplay/games/colors/ColorToolBox.java]: ```new AbstractGazeIndicator(...)``` ?

when a class is not ```abstract``` is should probably not be named ```Abstract...```

File: gazeplay/src/main/java/net/gazeplay/CurrentScreenPositionSupplierFactoryBean.java (hunk 1 above)
Comment (kuhlkrein) [reply gazeplay/src/main/java/net/gazeplay/CurrentScreenPositionSupplierFactoryBean.java]: Ah yes I remember I wanted to do that but I'm not used to the @Autowired spring annotation and I  wanted to ask you how we can declare only one instance of the CurrentScreenSupplier and share it with both of them by keeping the @Autowired ?..

Comment (YannRobert) [reply gazeplay/src/main/java/net/gazeplay/CurrentScreenPositionSupplierFactoryBean.java]: We can do this in a subsequent PR

Comment (kuhlkrein) [reply gazeplay/src/main/java/net/gazeplay/CurrentScreenPositionSupplierFactoryBean.java]: Ok fine, then the other requests have been fixed :)

**Pull request merged**